lib/sisimai/bite/email/google.rb

```diff
@@ -0,0 +1,320 @@
+module Sisimai::Bite::Email
+  module Google
+    # Sisimai::Bite::Email::Google parses a bounce email which created by Gmail.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Google.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/[@]googlemail[.]com[>]?\z/,
+        :subject => %r/Delivery[ ]Status[ ]Notification/,
+      }
+      Re1 = {
+        :begin   => %r/Delivery to the following recipient/,
+        :start   => %r/Technical details of (?:permanent|temporary) failure:/,
+        :error   => %r/The error that the other server returned was:/,
+        :rfc822  => %r{\A(?:
+             -----[ ]Original[ ]message[ ]-----
+            |[ \t]*-----[ ]Message[ ]header[ ]follows[ ]-----
+            )\z
+        }x,
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        expired: %r{(?:
+             DNS[ ]Error:[ ]Could[ ]not[ ]contact[ ]DNS[ ]servers
+            |Delivery[ ]to[ ]the[ ]following[ ]recipient[ ]has[ ]been[ ]delayed
+            |The[ ]recipient[ ]server[ ]did[ ]not[ ]accept[ ]our[ ]requests[ ]to[ ]connect
+            )
+        }x,
+        hostunknown: %r{DNS[ ]Error:[ ](?:
+             Domain[ ]name[ ]not[ ]found
+            |DNS[ ]server[ ]returned[ ]answer[ ]with[ ]no[ ]data
+            )
+        }x,
+      }
+      StateTable = {
```

Comment:
__[rubocop]__
* Style/MutableConstant: Freeze mutable objects assigned to constants.

<!-- Comment by [SideCI](https://sideci.com) -->